api/rest/restapi.go

```diff
@@ -384,8 +385,31 @@ func (api *API) addFileHandler(w http.ResponseWriter, r *http.Request) {
 		sendAcceptedResponse(w, errors.New("unsupported media type"))
 		return
 	}
+	outChan := make(chan *ipld.Node)
+	ctx, cancel := context.WithCancel(api.ctx)
+	go func() {
+		for nodePtr := range outChan {
```

Comment:
we want to stop reading from this channel if the api context is cancelled, so probably need to rewrite this with a `select`, to make sure we end the loop and return an error if we shutdown in the middle of this operation.